posts/2021-08-30-binding-app-to-postgresql-db-with-odo-and-ol-stack.adoc

```diff
@@ -0,0 +1,284 @@
+=== Bind the application to the PostgreSQL service
+
+List the available services to which the application can be bound. The PostgreSQL database service should be listed.
+
+[source,sh]
+----
+odo service list
+----
+
+Output:
+
+[source,sh]
+----
+NAME                        MANAGED BY ODO      STATE      AGE
+...
+Database/sampledatabase     Yes (mysboproj)     Pushed     50s
+...
+----
+
+Generate the service binding devfile configuration.
+
+[source,sh]
+----
+odo link Database/sampledatabase
+----
+
+Configuration added to devfile.yaml:
+
+[source,sh]
+----
+...
+- kubernetes:
+    inlined: |
+      apiVersion: binding.operators.coreos.com/v1alpha1
+      kind: ServiceBinding
+      metadata:
+        creationTimestamp: null
+        name: mysboproj-database-sampledatabase
+      spec:
+        application:
+          group: apps
+          name: mysboproj-app
+          resource: deployments
+          version: v1
+        bindAsFiles: false
+        detectBindingResources: true
+        services:
+        - group: postgresql.dev4devs.com
+          kind: Database
+          name: sampledatabase
+          version: v1alpha1
+      status:
+        secret: ""
+  name: mysboproj-database-sampledatabase
+...
+----
+
+Push the updates to the cluster. 
+
+[source,sh]
+----
+odo push
+----
+
+That is all. The application is now bound to the PostgreSQL database service. During the binding process, a secret containing the database connection information was created, and the pod hosting the application is restarted with the database connection information contained in the secret. The connection information is set in the application pod as environment variables.
+
+Let’s make sure we can use the application next.
+
+=== Use the application
+
+Find the URL to access the application through a browser.
+
+[source,sh]
+----
+odo url list
+----
+
+Output:
+
+[source,sh]
+----
+Found the following URLs for component mysboproj
+NAME     STATE      URL                                                                      PORT     SECURE     KIND
+ep1      Pushed     http://ep1-mysboproj-service-binding-demo.apps.my.os.cluster.ibm.com     9080     false      route
+----
+
+Open a browser and go to the URL shown by the previous step. Click "Create New Person" button.
+
+[.img_border_light]
+image::/img/blog/ol-stack-jpa-app-db-bind-browser-main.png[Main Page,width=70%,align="center"]
+
+Enter a user's name and age via the form shown on the page, and click  `Save`. The data is now persisted in the PostgreSQL database.
+
+[.img_border_light]
+image::/img/blog/ol-stack-jpa-app-db-bind-browser-data-entry.png[Data Input Page,width=70%,align="center"]
+
+After you save the data to the PostgreSQL database, notice that you are re-directed to the PersonList.xhtml page. The data being displayed was retrieved from the PostgreSQL database.
+
+[.img_border_light]
+image::/img/blog/ol-stack-jpa-app-db-bind-browser-show-data.png[Data Display Page,width=70%,align="center"]
+
```

Comment:
some kind of conclusion statement might be helpful, something like:

You just used the Service Binding Operator and to connect a sample application to a PostgGreSQL database.